language-support/hs/bindings/src/DA/Ledger/Services/PackageManagementService.hs

```diff
@@ -3,25 +3,66 @@
 
 {-# LANGUAGE GADTs #-}
 
-module DA.Ledger.Services.PackageManagementService (uploadDarFile) where
+module DA.Ledger.Services.PackageManagementService (
+    listKnownPackages, PackageDetails(..),
+    uploadDarFile,
+    ) where
 
-import Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService
-import Data.ByteString(ByteString)
+import DA.Ledger.Convert
 import DA.Ledger.GrpcWrapUtils
 import DA.Ledger.LedgerService
+import DA.Ledger.Types
+import Data.ByteString(ByteString)
+import Data.Text.Lazy (Text)
 import Network.GRPC.HighLevel.Generated
+import qualified Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService as LL
+
+data PackageDetails = PackageDetails
+    { pid :: PackageId
+    , size :: Int
```

Comment:
Might be worth considering to prefix these field names. `size` seems a bit too generic (yes, Haskell records suck).

Reply:
These short names correspond to the style adopted in:

https://github.com/digital-asset/daml/blob/master/language-support/hs/bindings/src/DA/Ledger/Types.hs

where it was an active decision to not uniquify field names, and instead use  `{-# LANGUAGE DuplicateRecordFields #-}` where necessary.

So far I am happy with this choice, but you may well disagree :)